lib/poda_payload.py

```diff
@@ -130,12 +143,18 @@ def get_data(self, vh):
         obj = ''
         try:
             obj = self.s3.Object(self.bucketname, vh).get()
+            return obj['Body'].read().decode('utf-8')
         except botocore.exceptions.ClientError as e:
             if e.response['Error']['Code'] == "404" or e.response['Error']['Code'] == 'NoSuchKey':
-                printdbg("Data does not exist for vh: %s" % vh)
-                return ''
+                tagData = self.storage_provider.getTagged(vh)
+                # The object does exist.
+                if (tagData.get("data") is None):
+                    printdbg("Data does not exist for vh: %s" % vh)
+                    return ''
+                else:
+                    cid = tagData.get("data").get("cid")
+                    data, _ = self.storage_provider.download(cid)
+                    return data.decode('utf-8')
             else:
                 # Something else has gone wrong.
                 raise
```

Comment:
should it raise or just return '', what does the API return if raise happens?

Reply:
You are correct. It always fetches from the lighthouse if the key is not found.

recall the original code snippet's logic used `raise.`
kindly refer to: https://github.com/syscoin/sentinel/blob/fe952557cc8e2067857c69bb194efb4660f50986/lib/poda_payload.py#LL139C1-L139C22

We simply mimicked the current logic with the proposed changes that were requested. We would modify the code as required